pkg/services/ngalert/store/alertmanager_test.go

```diff
@@ -275,6 +275,62 @@ func TestIntegrationAlertManagerConfigCleanup(t *testing.T) {
 	})
 }
 
+func TestMarkConfigurationAsApplied(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	sqlStore := db.InitTestDB(t)
+	store := &DBstore{
+		SQLStore: sqlStore,
+		Logger:   log.NewNopLogger(),
+	}
+
+	t.Run("attempting to mark a non existent config as applied shouldn't fail", func(tt *testing.T) {
+		cmd := models.MarkConfigurationAsAppliedCmd{
+			OrgID:             10,
+			ConfigurationHash: "test",
+		}
+		err := store.MarkConfigurationAsApplied(context.Background(), &cmd)
+		require.NoError(tt, err)
+	})
+
+	t.Run("marking an existent config should succeed", func(tt *testing.T) {
+		const orgID = 1
+		ctx := context.Background()
+
+		config, _ := setupConfig(t, "test", store)
+		err := store.SaveAlertmanagerConfiguration(ctx, &models.SaveAlertmanagerConfigurationCmd{
+			AlertmanagerConfiguration: config,
+			ConfigurationVersion:      "v1",
+			Default:                   false,
+			OrgID:                     orgID,
+		})
+		require.NoError(tt, err)
+
+		query := models.GetLatestAlertmanagerConfigurationQuery{
+			OrgID: orgID,
+		}
+		err = store.GetLatestAlertmanagerConfiguration(ctx, &query)
+		require.NoError(tt, err)
```

Comment:
```suggestion
		require.NoError(tt, err)
		require.Len(tt, query.Result, 0)
```
Let's assert that marking the config as applied actually did something.

Reply:
Why asserting that que query result length is zero? This method gets the latest config, which I later mark as applied.
Did you mean to make sure that there are no configs marked as applied at first?

Reply:
Yeah, it proves that MarkConfigAsApplied does something - we assert there are 0 applied configs, then mark, then assert that there is 1 applied config. The test enforces that something changed at that call (and not any time before).

Reply:
Sorry, I slightly misread this, I thought this was the result of GetAppliedConfigurations, not GetLatestAlertmanagerConfiguration. Let's:
- call GetAppliedConfigurations
- assert 0
- mark applied
- GetAppliedConfigurations again
- assert 1.